jina/peapods/pods/compoundpod.py

```diff
@@ -0,0 +1,253 @@
+__copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+import argparse
+from argparse import Namespace
+from typing import Optional, Dict, List, Union, Set
+
+from .. import Pea
+from .. import Pod
+from .. import BasePod
+from ...enums import PollingType, SchedulerType, PodRoleType, PeaRoleType
+
+
+class CompoundPod(BasePod):
+    """A CompoundPod is a immutable set of pods, which run in parallel. They share the same input and output socket.
+    Internally, the peas of the pods can run with the process/thread backend. They can be also run in their own containers.
+    :param args: arguments parsed from the CLI
+    :param needs: pod names of preceding pods, the output of these pods are going into the input of this pod
+    """
+
+    def __init__(self, args: Union['argparse.Namespace', Dict], needs: Set[str] = None):
+        super().__init__(args, needs)
+        self.replica_list = []  # type: List['Pod']
+        if isinstance(args, Dict):
+            # This is used when a Pod is created in a remote context, where peas & their connections are already given.
+            self.replicas_args = args
+        else:
+            self.replicas_args = self._parse_args(args)
+
+    @property
+    def first_pea_args(self) -> Namespace:
```

Comment:
This should not be the first pea of the first replica. As for the Flow is concerned, this should be the arguments of the HeadPea of the replicas. Maybe the naming of these properties need to change

Reply:
Also here u are mixing pea naming with what is supposed to be Pod arguments

Reply:
thanks, I fixed the naming